docs/adrs/003-bugzilla-response-codes.md

```diff
@@ -0,0 +1,100 @@
+# Bugzilla disables webhooks after too many errors
+
+- Status: proposed
+- Date: 2023-27-11
+
+Tracking issue: 
+
+## Context and Problem Statement
+When bugzilla receives too many error responses from JBI, it stops triggering webhook calls for the entire project causing data to stop syncing. Frequently these errors are due to JBI being unable to process a payload due to errors in configuration (or incomplete configuration) in Jira or a mismatch of data for a single bug. 
+
+We don't want the entire sync process to stop because of this. We have identified four options to solve this problem.
+
+## Decision Drivers
+
+- Amount of initial engineering effort
+- Amount of maintenance effort
+- Overall performance of JBI (how quickly is data able to move)
+- How intuitive the solution is to the users that depend on the data (will picking the easiest option solve their needs?)
+
+## Considered Options
+For all of these options, we will be returning a succesful response to bugzilla's webhook calls. I propose a 202 status code to identify that we're accepting the request immediately and are processing it asynchronously.
```

Comment:
We might be limited by Bugzilla here. ([docs](https://bugzilla.mozilla.org/page.cgi?id=webhooks.html))

> If the response is not 200, the Bugzilla system will retry 4 attempts, waiting 5s, 10s, 15s and 20s between each attempt. If none of those attempts is successful, the system will continue trying every 15 minutes.

Reply:
I was really hoping they just checked if the response status is OK, but you're right they're intentionally checking for 200. Will update the ADR.
https://github.com/bugzilla/harmony/blob/e72338d1a8c374e0f1ee88770a05701067fa6b64/extensions/Push/lib/Connector/Webhook.pm#L113